File: containsDuplicates.js
// Before Video Solution
// Results: Runtime 5766ms - Beat 6.77% | Memory 51.88 - Beat 83.22%
function containsDuplicates(nums){
    const numArray = [...nums]
    while(numArray.length > 0){
        const numberToCheck = numArray.pop()
        if(numArray.indexOf(numberToCheck) > -1){
            return true
        }
    }
    return false
}

// Solution After Video (from memory)
// Results: Runtime 138ms - Beat 28.00% | Memory 51.51 - Beat 87.17%
function containsDuplicatesRev(nums){
    nums.sort((a, b) => a-b)
    for (let i=0; i<nums.length-1; i++){
        if(nums[i] === nums[i+1]){
            return true
        }
    }
    return false
}


// Solution After Video & Research into Set
// Results: Runtime 77ms - Beat 74.51% | Memory 52.68 - Beat 72.64%
function containsDuplicatesSet(nums){
    const numSet = new Set()
    for(let i=0; i<nums.length; i++){
        if(numSet.has(nums[i])){
            return true
        }
        numSet.add(nums[i])
    }
    return false
}

File: contains_duplicates.py
# Solution attempt after watching javascript video before python video
# Results: Runtime 483ms - Beat 38.62% | Memory 30.78 - Beat 88.56%
def containsDuplicate(nums):
    return len(set(nums)) != len(nums)

# Solution attempt after watching video
# Results: Runtime 477ms - Beat 54.54% | Memory 31.78 - Beat 38.45%
def postContainsDuplicate(nums):
    numsHashset = set()
    for n in nums:
        if n in numsHashset:
            return True
        numsHashset.add(n)
    return False


